sky/compositor/picture_layer.cc

```diff
@@ -18,10 +26,30 @@ void PictureLayer::Paint(PaintContext::ScopedFrame& frame) {
   DCHECK(picture_);
 
   SkCanvas& canvas = frame.canvas();
-  canvas.save();
-  canvas.translate(offset_.x(), offset_.y());
-  canvas.drawPicture(picture_.get());
-  canvas.restore();
+
+#if ENABLE_RASTER_CACHE
+  const SkMatrix& ctm = canvas.getTotalMatrix();
+  SkISize size = SkISize::Make(paint_bounds().width() * ctm.getScaleX(),
```

Comment:
are we sure this is the exact size that drawPicture() will rasterise at?
Consider a 10x10 rectangle rotated 45 degrees. getScaleX() is going to return something like 0.5, not 1.4, no?